src/sentry/nodestore/riak/__init__.py

```diff
@@ -8,3 +8,44 @@
 from __future__ import absolute_import
 
 from .backend import *  # NOQA
+
+
+# HACK: Monkey patch the RiakPbcConnection client
+# to enable TCP keepalives. This is to see if it fixes
+# out connection timeout issues. If this works, ideally
+# we contribute upstream
+def patch_socket_keepalive():
+    import socket
+
+    from riak.transports.pbc.connection import RiakPbcConnection
+
+    original_connect = RiakPbcConnection._connect
+
+    def patched_connect(self):
+        original_connect(self)
+        s = self._socket
+
+        if not s:
+            return
+
+        # Future-proof ourselves against the socket getting keepalive set already
+        if not s.getsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE):
+            return
+
+        # Enable TCP keepalive
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
+
+        # Make sure we have TCP_KEEPIDLE before setting it.
+        if hasattr(socket, 'TCP_KEEPIDLE'):
+            # overrides value (in seconds) shown by sysctl net.ipv4.tcp_keepalive_time
+            s.setsockopt(socket.SOL_TCP, socket.TCP_KEEPIDLE, 4 * 60)
```

Comment:
We can, but not sure if we want to blindly apply to every TCP connection, unless @mitsuhiko thinks that's fine.